### PR TITLE
Run3-alca216Y To make the response study work without HLT information if needed

### PR DESCRIPTION
**Pull request by bsunanda**
#### PR description:

To make the response study work without HLT information if needed

#### PR validation:

Used in the new validation of Geant4 package

#### if this PR is a backport please specify the original PR and why you need to backpaort that PR:

Nothing special

### CODE REVIEW

**Comment by bsunanda** [comment]
@cmsbuild Please test

File: Calibration/IsolatedParticles/plugins/StudyCaloResponse.cc [hunk 1]
```diff
@@ -419,16 +435,20 @@ void StudyCaloResponse::analyze(edm::Event const& iEvent, edm::EventSetup const&
...
-    for (int ind = 0; ind < nvtxs; ind++) {
-      if (!((*recVtxs)[ind].isFake()) && (*recVtxs)[ind].ndof() > 4)
-        ngoodPV++;
-    }
-    for (int i = 0; i < nPVBin_; ++i) {
-      if (ngoodPV >= pvBin_[i] && ngoodPV < pvBin_[i + 1]) {
-        nPV = i;
-        break;
+    int ntrk(0), ngoodPV(0), nPV(-1), nvtxs(0);
+    if (!recVtxs.isValid()) {
```
Comment (tvami) [inline Calibration/IsolatedParticles/plugins/StudyCaloResponse.cc:439]: Shouldnt you just not run this when there is no valid vertex? Give an exception or something?

Comment (bsunanda) [reply Calibration/IsolatedParticles/plugins/StudyCaloResponse.cc:439]: Doing that

**Comment by tvami** [comment]
@cmsbuild , please test

**Comment by tvami** [comment]
@cmsbuild , please test

File: Calibration/IsolatedParticles/plugins/StudyCaloResponse.cc [hunk 2]
```diff
@@ -328,7 +328,11 @@ void StudyCaloResponse::analyze(edm::Event const& iEvent, edm::EventSetup const&
   bool ok(false);
   std::string triggerUse("None");
   if (!triggerEventHandle.isValid()) {
-    edm::LogWarning("IsoTrack") << "Error! Can't get the product " << triggerEvent_.label();
+    if (trigNames_.empty()) {
+      ok = true;
+    } else {
+      edm::LogWarning("IsoTrack") << "Error! Can't get the product " << triggerEvent_.label();
```
Comment (tvami) [inline Calibration/IsolatedParticles/plugins/StudyCaloResponse.cc:334]:
```suggestion
      edm::LogWarning("StudyCaloResponse") << "Error! Can't get the product " << triggerEvent_.label();
```

Sorry, I just noticed this, the whole file is using `edm::LogWarning("IsoTrack")` but the plugin name is  `StudyCaloResponse`, so it should be that passed to the `LogWarning`

**Comment by tvami** [comment]
@cmsbuild , please test

**Comment by bsunanda** [comment]
@cmsbuild Please test

**Comment by tvami** [comment]
+alca
 * diffs in MsgLogger only

**Pull request merged**